content/en/getting_started/code_analysis/_index.md

```diff
@@ -302,6 +308,72 @@ Click on a service to access information about CI pipelines from Pipeline Visibi
 
 {{< img src="/getting_started/code_analysis/catalog_service.png" alt="A link to the source code directly in GitHub from a detected library vulnerability" style="width:100%" >}}
 
+### Linking services to code violations and libraries
+
+Datadog associates code violations or libraries with relevant services by using the following mechanisms:
+
+1. [Identifying the code location associated with a service using the Service Catalog.](#identifying-the-code-location-in-the-service-catalog)
+2. [Detecting usage patterns of files within additional Datadog products.](#detecting-file-usage-patterns)
+3. [Searching for the service name in the file path or repository.](#detecting-service-name-in-paths-and-repository-names)
+
+If one method succeeds, no further matching attempts are made. Each mapping method is detailed below.
+
+#### Identifying the code location in the Service Catalog
+
+The schema version `v3` and later of the Service Catalog allows you
+to add the mapping of your code location for your service. The `codeLocations`
+section specifies the locations of the code with the repository that
+contains the code and its associated `paths`.
+
+The `paths` attribute is a list of [globs][14]
+that should match paths in the repository.
+
+{{< code-block lang="yaml" filename="service.datadog.yaml" collapsible="true" >}}
+apiVersion: v3
+kind: service
+metadata:
+  name: my-service
+datadog:
+  codeLocations:
+    - repositoryURL: https://github.com/myorganization/myrepo.git
+      paths:
+        - /path/to/service/code/**
+{{< /code-block >}}
+
+
+#### Detecting file usage patterns
+
+Datadog detects file usage in additional products such as Error Tracking and associate
+files with the runtime service. For example, if a service called `foo` has
+a log entry or a stack trace containing a file with a path `/modules/foo/bar.py`,
+it associates files `/modules/foo/bar.py` to service `foo`.
+
+#### Detecting service name in paths and repository names
+
+Datadog detects service name in paths and repository names and associates
+the file with the service if it finds a match.
+
+For a repository match, if there is a service called `myservice` and
+the repository URL is `https://github.com/myorganization/myservice.git`, then,
+it associates `myservice` to all files in the repository.
+
+If no repository match is found, Datadog attempts to find a match in the
+`path` of the file. If there is a service `myservice` and the path is
+`/path/to/myservice/foo.py`, it associates the file with `myservice` because
+the name of the service is part of the path. If two services are present
```

Comment:
```suggestion
`path` of the file. If there is a service named `myservice`, and the path is `/path/to/myservice/foo.py`, the file is associated with `myservice` because the service name is part of the path. If two services are present
```
Re-wording this one a bit too.